app.py

```diff
@@ -142,6 +156,882 @@ def viewOptions():
+    if isNonEmptyQuery(query) == False:
+        print("User doesn't belong to the group")
+        return
+    try:
+        query = "INSERT INTO IS_ADMIN VALUES(%s, %s);" (row["user_id"], row["group_id"])
+        cur.execute(query)
+        con.commit()
+    except Exception as e:
+        con.rollback()
+        print(e)
+        print("Error: Check your inputs.")
+
+def makeUserModerator():
+    global cur
+    row = {}
+    row["user_id"] = input("Enter the ID of the user to make him an moderator of a group: ")
+    row["group_id"] = input("Enter the ID of the group for which user should be made an moderator of: ")
+    # Don't we have to check if the user is a member of the group?
+    query = "SELECT * FROM BELONGS_TO where group_id=%s and user_id=%s;" % (row["group_id"], row["user_id"])
+    if isNonEmptyQuery(query) == False:
+        print("User doesn't belong to the group")
+        return
+    try:
+        query = "INSERT INTO IS_MODERATOR VALUES(%s, %s);" (row["user_id"], row["group_id"])
+        cur.execute(query)
+        con.commit()
+    except Exception as e:
+        con.rollback()
+        print(e)
+        print("Error: Check your inputs.")
+
+def makeReactionToAComment():
+    global cur
+    row = {}
+    row["user_id"] = input("Enter the ID of the user who made the reaction: ")
+    row["comment_id"] = input("Enter the ID of the comment in which the reaction was made: ")
+    print("Choose the react type by pressing the corresponding number")
+    print("1 . Like")
+    print("2 . Dislike")
+    print("3 . Wow")
+    print("4 . Heart")
+    print("5 . Angry")
+    print("6 . Haha")
+    reactNum = 123
+    try:
+        reactNum = int(input())
+    except:
+        print("Invalid react Type")
+
+    if reactNum == 1:
+        row["reactedType"]="Like"
+    elif reactNum == 2:
+        row["reactedType"]="Dislike"
+    elif reactNum == 3:
+        row["reactedType"]="Wow"
+    elif reactNum == 4:
+        row["reactedType"]="Heart"
+    elif reactNum == 5:
+        row["reactedType"]="Angry"
+    elif reactNum == 6:
+        row["reactedType"]="Haha"
+    else :
+        print("Invalid react Type")
+        return
+    
+    try:
+        query = "INSERT INTO MAKES_A_REACT(comment_id, user_id, reacted_type) VALUES(%s, %s, '%s');" % (
+            row["comment_id"], row["user_id"],row["reactedType"])
+        cur.execute(query)
+        con.commit()
+    except Exception as e:
+        con.rollback()
+        print(e)
+        print("Error: Check your inputs.")
+
+def mentionInComment():
+    global cur
+    row = {}
+    row["comment_id"] = input("Enter the ID of the comment: ")
+    row["mentioner_id"] = input("Enter the ID of the user who mentioned someone: ")
+    row["mentionee_id"] = input("Enter the ID of the user who got mentioned: ")
+    try:
+        query = "INSERT INTO MENTIONS VALUES(%s, %s, %s);" (row["mentioner_id"], row["mentionee_id"], row["comment_id"])
```

Comment:
Have to check if the mentioner_id is in fact the ID of the person who made this comment.

Reply:
Done.